packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -71,12 +72,23 @@ const EditWithGeneratedProps = ( props ) => {
 		return <EditWithFilters { ...props } context={ context } />;
 	}
 
-	// Generate a class name for the block's editable form.
-	const generatedClassName = hasBlockSupport( blockType, 'className', true )
-		? getBlockDefaultClassName( name )
-		: null;
+	const generatedClassNames = [];
+
+	if ( hasBlockSupport( blockType, 'className', true ) ) {
+		generatedClassNames.push( getBlockDefaultClassName( name ) );
+
+		const activeVariation = getActiveBlockVariation( name, attributes );
+		if ( activeVariation && activeVariation?.name ) {
+			generatedClassNames.push(
+				getBlockDefaultClassName(
+					`${ name }/${ activeVariation.name }`
+				)
```

Comment:
Per https://github.com/WordPress/gutenberg/pull/61864/files#r1655059595, let's maybe change this to

```suggestion
				getBlockDefaultClassName( name ) + `-${ activeVariation.name }`
```

(There are probably a few more instances of this in this PR.)